core/dbt/utils.py

```diff
@@ -599,3 +603,22 @@ def __getitem__(self, name: str) -> Any:
 
     def __contains__(self, name) -> bool:
         return any((name in entry for entry in self._itersource()))
+
+
+def _exception_retry(fn):
+    @functools.wraps(fn)
+    def wrapper(*args, **kwargs):
+        max_attempts = 5
+        attempt = 0
+        while True:
+            attempt += 1
+            try:
+                return fn(*args, **kwargs)
+            except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as exc:
+                if attempt < max_attempts:
+                    time.sleep(1)
```

Comment:
Do you think we might want to do some logging? Something like: logger.debug(f"Attempt {attempt} to connect.").